eclipse/plugins/org.switchyard.tools.ui/src/org/switchyard/tools/ui/common/SwitchYardSettingsGroup.java

```diff
@@ -805,7 +805,15 @@ private void populateRuntimeVersionsList(IProgressMonitor monitor) {
         }
         // add default version
         final ArtifactVersion defaultVersion = parseVersion(NewSwitchYardProjectWizard.DEFAULT_RUNTIME_VERSION);
-        _availableVersions.add(defaultVersion);
+        if (!_availableVersions.contains(defaultVersion)) {
+            _availableVersions.add(defaultVersion);
+        }
+
+        // for SWITCHYARD-2959 - add previous version as well 
```

Comment:
Is that a stylistic thing? Because you didn't comment on it in the changes for https://github.com/jboss-switchyard/tools/pull/637

If it is a style thing and something we want to achieve going forward, I'm fine with it - but I personally like leaving breadcrumbs back to the reasons behind why certain changes were made. The goal of comments is to make the code more readable and this just gives us more information. Just my opinion of course.